Source/AppAuth/macOS/OIDAuthState+Mac.h

```diff
@@ -35,11 +36,18 @@ NS_ASSUME_NONNULL_BEGIN
         and update the OIDAuthState with the results (@c
         OIDAuthState.updateWithTokenResponse:error:).
     @param authorizationRequest The authorization request to present.
+    @param presentingWindow The window to present the authentication flow.
     @param callback The method called when the request has completed or failed.
     @return A @c OIDExternalUserAgentSession instance which will terminate when it
         receives a @c OIDExternalUserAgentSession.cancel message, or after processing a
         @c OIDExternalUserAgentSession.resumeExternalUserAgentFlowWithURL: message.
  */
++ (id<OIDExternalUserAgentSession>)
+    authStateByPresentingAuthorizationRequest:(OIDAuthorizationRequest *)authorizationRequest
+                             presentingWindow:(NSWindow *)presentingWindow
+                                     callback:(OIDAuthStateAuthorizationCallback)callback
+                                     API_AVAILABLE(macosx(10.15));
```

Comment:
Instead of making this available only in 10.15+, we should consider having it be the preferred path for all versions and deprecate the method below.  This would be similar to what has been done for iOS.  We want to encourage developers to make use of ASWebAuthentiationSession when available as this is a better experience.

Reply:
Done.

Reply:
Note that you're still restricting this method to 10.15+.  I'm proposing that developers be able to use this method for any version of macOS, with the implementation determining which auth mechanism to use (as it already does).  This would resemble our approach for iOS.

Reply:
Done.